### PR TITLE
Making render(admin)?(header|footer) awaitable

### PR DESCRIPTION
**Pull request by julianlam**

### CODE REVIEW

File: src/middleware/admin.js [hunk 1]
```diff
@@ -37,93 +39,69 @@ module.exports = function (middleware) {
...
+			bodyClass: data.bodyClass,
+			version: version,
+			latestVersion: results.latestVersion,
+			upgradeAvailable: results.latestVersion && semver.gt(results.latestVersion, version),
+		};
+
+		templateValues.template = { name: res.locals.template };
+		templateValues.template[res.locals.template] = true;
+
+		const render = util.promisify(req.app.render.bind(req.app));
```
Comment (barisusakli) [inline src/middleware/admin.js:92]: Should be able to use `req.app.renderAsync`

File: src/middleware/admin.js [hunk 2]
```diff
@@ -37,93 +39,69 @@ module.exports = function (middleware) {
...
+		return scripts.map(function (script) {
+			return { src: script };
+		});
 	}
 
-	middleware.admin.renderFooter = function (req, res, data, next) {
-		req.app.render('admin/footer', data, next);
+	middleware.admin.renderFooter = async function (req, res, data) {
+		const render = util.promisify(req.app.render.bind(req.app));
+		return await render('admin/footer', data);
```
Comment (barisusakli) [inline src/middleware/admin.js:105]: same as above `req.app.renderAsync` already available

File: src/middleware/admin.js (hunk 1 above)
Comment (julianlam) [reply src/middleware/admin.js:92]: Oh, I didn't know that, thanks 👍

File: src/middleware/admin.js (hunk 2 above)
Comment (julianlam) [reply src/middleware/admin.js:105]: 👍

File: src/middleware/header.js [hunk 3]
```diff
@@ -73,207 +74,164 @@ module.exports = function (middleware) {
...
+
+		results.user.email = String(results.user.email);
+		results.user['email:confirmed'] = results.user['email:confirmed'] === 1;
+		results.user.isEmailConfirmSent = !!results.isEmailConfirmSent;
+
+		templateValues.bootswatchSkin = (parseInt(meta.config.disableCustomUserSkins, 10) !== 1 ? res.locals.config.bootswatchSkin : '') || meta.config.bootswatchSkin || '';
+		templateValues.config.bootswatchSkin = templateValues.bootswatchSkin || 'noskin';	// TODO remove in v1.12.0+
+
+		const unreadCounts = results.unreadData.counts;
+		var unreadCount = {
```
Comment (pitaj) [inline src/middleware/header.js:122]: Switch from var to let or const

**Pull request merged**